code/__HELPERS/unsorted.dm

```diff
@@ -1594,6 +1594,15 @@ atom/proc/GetTypeInAllContents(typepath)
 			colour += temp_col
 	return colour
 
+/proc/get_random_chemical(var/is_plant = 0)
+	var/list/blocked = blocked_chems		//blocked_chems list is found in code/_globalvars/lists/reagents.dm

```

Comment:
depending on how in/sane byond is with lists, it could potentially just continually expand blocked_chems with the plant_blocked_chems, therefore making any call to it prevent plant_blocked_chems